### ISSUE TITLE
✨ [Feature] The guard driver of Auth support custom driver

### ISSUE THREAD

**Issue by hwbrzzl** [goravel/goravel#478]
### Before feedback (在反馈之前)

- [X] There are no features that I want to submit in Issues (当前 Issues 中没有我将要提交的新功能)

### Describe Feature (描述功能)

Reference: https://laravel.com/docs/11.x/authentication#adding-custom-guards

**Comment by praem90** [comment]
@hwbrzzl Laravel uses `AuthManager` to create the `guard` and the `userProvider`. 

Those three has to be implemented here to achieve Laravel's Authentication without breaking the existing behavior .

What are your thoughts on this?

**Comment by hwbrzzl** [comment]
Sorry, I did get you, for example?

**Comment by praem90** [comment]
Laravel uses the Manager pattern to register and retrive the `AuthGuards`.

https://laravel.com/api/11.x/Illuminate/Auth/AuthManager.html

The `UserProvider` is decoupled from the `AuthGuard`. 
All the Guards must accept the `UserProvider` to retrive the user. 

Guards:
https://laravel.com/api/11.x/Illuminate/Auth/SessionGuard.html#method___construct
https://laravel.com/api/11.x/Illuminate/Auth/TokenGuard.html#method___construct

UserProvider:
https://laravel.com/api/11.x/Illuminate/Auth/DatabaseUserProvider.html#method___construct
https://laravel.com/api/11.x/Illuminate/Auth/EloquentUserProvider.html#method___construct


The Auth library in Goravel is tightly coupled with `jwt` and `orm`. To refactor this to match laravel's Authendication,
I would like to contribute to goravel by introducing `AuthManager`, `AuthGuard` and `UserProvider` .

Would you like this approach or do you have any other suggestions?

**Comment by hwbrzzl** [comment]
Yes, AuthManager, AuthGuard and UserProvider are a great structure, we can have it. If you can contribute, I appreciate that. 👍

**Comment by praem90** [comment]
Great, I would love to contribute ❤️

**Comment by saqijaan** [comment]
@praem90 thanks in advance for your contribution. I was working on this and but due to existing pattern I just override the existing JWT auth implementation with sanctum. I was little bit busy with other projects. I'll back soon let me know if you need any help. Waiting on your implementation PR. ❤️

**Comment by praem90** [comment]
Thanks @saqijaan. Sure, I'll keep posting my updates on this thread.

**Comment by praem90** [comment]
I've just getting started with `AuthManager` and the work is still in progress. But in the meantime I just wanted to know your feedback on my implementation and the best practices. 

https://github.com/praem90/goravel-framework/tree/feat-478-custom-auth-guard

**Comment by praem90** [comment]
Also I wanted to know the upstream branch because there is no `main` or `master` branch.

**Comment by hwbrzzl** [comment]
The master branch
